transports/p2p/connmgr/connmanager.go

```diff
@@ -209,17 +253,25 @@ func (cm *ConnManager) handleFailedConn(c *ConnReq) {
 			cm.Connect(c)
 		})
 	} else if cm.cfg.GetNewAddress != nil {
-		cm.failedAttempts++
-		if cm.failedAttempts >= maxFailedAttempts {
-			cm.log.Debug().Msgf("Max failed connection attempts reached: [%d] "+
-				"-- retrying connection in: %v", maxFailedAttempts,
-				cm.cfg.RetryDuration)
-			time.AfterFunc(cm.cfg.RetryDuration, func() {
-				cm.NewConnReq()
-			})
+		if c.Addr != nil && c.Addr.String() != "" && cm.cfg.BanAddress != nil {
+			cm.incAddrConnAttempt(c.Addr.String())
```

Comment:
Why we don't increment global connection fails here also?

Reply:
From my understanding of the code, the global connection failure counter was there because of the connection requests without addresses (that I mentioned before). Without it we have an almost endless loop of connecting until a valid address pops up. But when it comes to connection requests with addresses, there's no need to slow it down, because after 25 unsuccessful attempts, they're gonna be banned anyways (with this solution) and we don't need to block the entire connManager. But if you feel like we should, I can change it :)

Reply:
ok, lets leave it this way, at least until we find global solution for those empty addresses